`make shell` failing to start on OS X

When executing make-shell on macos we are getting the following error:

```
bash-4.4$ make shell
Creating virtualenv...
Installing Python dependencies...
osc-lib 1.9.0 requires Babel!=2.4.0,>=2.3.4, which is not installed.
osc-lib 1.9.0 requires cliff!=2.9.0,>=2.8.0, which is not installed.
osc-lib 1.9.0 requires keystoneauth1>=3.3.0, which is not installed.
osc-lib 1.9.0 requires openstacksdk>=0.9.19, which is not installed.
osc-lib 1.9.0 requires os-client-config>=1.28.0, which is not installed.
osc-lib 1.9.0 requires oslo.i18n>=3.15.3, which is not installed.
osc-lib 1.9.0 requires oslo.utils>=3.33.0, which is not installed.
osc-lib 1.9.0 requires simplejson>=3.5.1, which is not installed.
osc-lib 1.9.0 requires stevedore>=1.20.0, which is not installed.
python-heatclient 1.14.0 requires Babel!=2.4.0,>=2.3.4, which is not installed.
python-heatclient 1.14.0 requires cliff!=2.9.0,>=2.8.0, which is not installed.
python-heatclient 1.14.0 requires iso8601>=0.1.11, which is not installed.
python-heatclient 1.14.0 requires keystoneauth1>=3.3.0, which is not installed.
python-heatclient 1.14.0 requires oslo.i18n>=3.15.3, which is not installed.
python-heatclient 1.14.0 requires oslo.serialization!=2.19.1,>=2.18.0, which is not installed.
python-heatclient 1.14.0 requires oslo.utils>=3.33.0, which is not installed.
python-heatclient 1.14.0 requires PrettyTable<0.8,>=0.7.1, which is not installed.
python-swiftclient 3.5.0 requires futures>=3.0; python_version == "2.7" or python_version == "2.6", which is not installed.
elasticsearch 6.0.0 has requirement urllib3<1.23,>=1.21.1, but you'll have urllib3 1.23 which is incompatible.
Downloading kubectl...
Downloading Helm...
/bin/sh: -c: line 1: syntax error: unexpected end of file
make: *** [shell] Error 2
```

Looking at the commit that introduced this bug we found the following one: 

```
MacBook-Pro-de-Thomas:metal-k8s thomas$ git bisect good
06c5bbc5dc41afec40f1ce1dd40a089cc7f3984f is the first bad commit
commit 06c5bbc5dc41afec40f1ce1dd40a089cc7f3984f
Author: Nicolas Trangez <ikke@nicolast.be>
Date:   Tue Jun 19 16:59:27 2018 +0200

    Makefile: Add `C` option to `shell` target

:100644 100644 3d072e3aa6a56ad7fe052ef809d88925abc9c6a0 c862ccdfd54980652bed9e1061b5e337b215c5fd M	Makefile
```

--- a/ChangeLog.rst
+++ b/ChangeLog.rst
@@ -27,6 +27,8 @@ Bugs fixed
 :ghissue:`381` - warn when Bash completion is not available in `make shell`
 (:ghpull:`382`)
 
+:ghissue:`192` - `make shell` failing to start on OS X (:ghpull:`418`)
+
 Release 1.0.0
 =============
 This marks the first production-ready release of `MetalK8s`_. Deployments using

--- a/Makefile
+++ b/Makefile
@@ -97,7 +97,7 @@ $(KUBECTL_BIN): $(SHELL_ENV_EXISTS)
 	$(V)mv $@.tmp $@
 
 KUBECTL = $(VENV_BIN)/kubectl
-KUBECTL_REALPATH = realpath $(KUBECTL_BIN)
+KUBECTL_REALPATH = $(PYTHON) hack/realpath.py $(KUBECTL_BIN)
 $(KUBECTL): $(KUBECTL_BIN) $(VENV_EXISTS)
 	$(V)ln -sf $(shell $(KUBECTL_REALPATH))$(KUBECTL_REALPATH:sh) $@
 
@@ -119,7 +119,7 @@ $(HELM_BIN): $(HELM_SRC_TAR) $(SHELL_ENV_EXISTS)
 	$(V)touch $@
 
 HELM = $(VENV_BIN)/helm
-HELM_REALPATH = realpath $(HELM_BIN)
+HELM_REALPATH = $(PYTHON) hack/realpath.py $(HELM_BIN)
 $(HELM): $(HELM_BIN) $(VENV_EXISTS)
 	$(V)ln -sf $(shell $(HELM_REALPATH))$(HELM_REALPATH:sh) $@
 

--- a/Makefile
+++ b/Makefile
@@ -130,17 +130,17 @@ $(BASHRC): $(SHELL_ENV_EXISTS) hack/shell-bashrc
 
 shell: $(VENV_EXISTS) $(REQUIREMENTS_INSTALLED) $(KUBECTL) $(HELM) $(BASHRC) ## Run a shell with `ansible-playbook`, `kubectl` and `helm` pre-installed
 	$(V)BASH_ENV="$(BASHRC)"; export BASH_ENV; if test -z "$(C)"; then \
-		# Interactive shell \
+		`# Interactive shell` \
 		echo "Launching MetalK8s shell environment. Run 'exit' to quit."; \
 		bash --rcfile $(BASHRC) ||:; \
 	elif test "x$(C)" = "x-"; then  \
-		# Semi-interactive shell, keeping exit-code \
+		`# Semi-interactive shell, keeping exit-code` \
 		bash --rcfile $(BASHRC); \
 	elif test -e "$(C)"; then \
-		# A script \
+		`# A script` \
 		bash --rcfile $(BASHRC) "$(C)"; \
 	else \
-		# A command \
+		`# A command` \
 		bash --rcfile $(BASHRC) -c "$(C)"; \
 	fi
 .PHONY: shell

--- a/hack/realpath.py
+++ b/hack/realpath.py
@@ -1,0 +1,15 @@
+#!/usr/bin/env python
+
+import optparse
+import os.path
+
+
+def realpath(files):
+    for file_ in files:
+        print(os.path.realpath(file_))
+
+
+if __name__ == '__main__':
+    parser = optparse.OptionParser()
+    (_, pos_args) = parser.parse_args()
+    realpath(pos_args)

